src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/DetailedCalculationResultsService.kt

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
+
+import jakarta.persistence.EntityNotFoundException
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.entity.CalculationRequest
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.exceptions.BreakdownChangedSinceLastCalculation
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.Booking
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.BreakdownMissingReason
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.DetailedCalculationResults
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.repository.CalculationRequestRepository
+
+@Service
+@Transactional(readOnly = true)
+open class DetailedCalculationResultsService(
```

Comment:
This started in CalculationTransactionalService but the tests do not mock any of the data so it was very difficult and verbose to write the error handling tests. CalculationTransactionalServiceTest actually runs a proper calc when you generate a breakdown...